src/notifications/notificationmanager.cpp

```diff
@@ -103,6 +105,9 @@ NotificationManager::NotificationManager(QObject *parent) :
     databaseCommitTimer.setSingleShot(true);
     connect(&databaseCommitTimer, SIGNAL(timeout()), this, SLOT(commit()));
 
+    expirationTimer.setSingleShot(true);
+    connect(&expirationTimer, SIGNAL(timeout()), this, SLOT(expire()));
+
```

Comment:
IIRC, it will just freeze, so it should be triggered, but where it starts counting again is unpredictable.